src/mlpack/bindings/python/mlpack/matrix_utils.py

```diff
@@ -81,7 +81,10 @@ def to_matrix_with_info(x, dtype, copy=False):
 
   if isinstance(x, np.ndarray):
     # It is already an ndarray, so the vector of info is all 0s (all numeric).
-    d = np.zeros([x.shape[1]], dtype=np.bool)
+    if len(x.shape) < 2:
+      d = np.zeros(0, dtype=np.bool)
```

Comment:
Shouldn't this just be 1 in this case though? Correct me if I'm wrong.  This might work and run, but I suspect that there is an invalid memory access from the mlpack side if `d` really has length 0.  This comment applies to the other `d = np.zeros(0, dtype=np.bool)` too.

Reply:
Hi @rcurtin I was also bit confused about this. Thank you for correcting me. :)